performance-storage-service/pss_project/api/models/database/OLTPBenchResult.py

```diff
@@ -24,3 +30,65 @@ class Meta:
     max_connection_threads = PositiveSmallIntegerField()
     metrics = JSONField(encoder=DjangoJSONEncoder)
     incremental_metrics = JSONField(encoder=DjangoJSONEncoder)
+
+    def get_test_config(self):
+        """ Return a dict with the fields related to the OLTPBench test config """
+        config = {}
+        for field in PERFORMANCE_CONFIG_FIELDS:
+            config[field] = getattr(self, field)
+        return config
+
+    def get_oltpbench_config_query(self, branch):
+        """ Returns a query set for a branch based on the current object's config. This is used to create a query to
+        compare similar test results across branches. """
+        filters = {
+            "git_branch": branch,
+            "time__gte": datetime.now(timezone.utc)-timedelta(days=7),
+        }
+        for field in PERFORMANCE_CONFIG_FIELDS:
+            filters[field] = getattr(self, field)
+        return OLTPBenchResult.objects.filter(**filters)
+
+    def is_config_match(self, oltpbench_result):
+        """ Checks whether another OLTPBenchResult object has a matching config to the current object. """
+        for field in PERFORMANCE_CONFIG_FIELDS:
+            if getattr(self, field) != getattr(oltpbench_result, field):
+                return False
+        return True
+
+    def compare_throughput(self, oltpbench_result):
+        """ Calculate the % difference between the current OLTPBenchResult object and another OLTPBenchResult object.
+        """
+        config = {}
+        for field in PERFORMANCE_CONFIG_FIELDS:
+            config[field] = getattr(self, field)
+        base_throughput = float(self.metrics.get('throughput', 0))
+        if base_throughput == 0:
+            return 0
+        new_throughput = float(oltpbench_result.metrics.get('throughput', 0))
+        percent_diff = (new_throughput - base_throughput) / base_throughput * 100
+        return percent_diff
+
+    @classmethod
+    def get_all_branch_results(cls, branch):
+        """ Return a query set that will return all results for a given branch. """
+        branch_results = cls.objects.filter(git_branch=branch)
+        return branch_results
+
+    @classmethod
+    def get_latest_branch_results(cls, branch):
+        """ Return a query set that will return all the latest OLTPBenchResults for each unique test config in a given
+        branch. """
+        branch_results = cls.objects.filter(git_branch=branch, time__gte=datetime.now(
+            timezone.utc)-timedelta(days=7)).order_by(*PERFORMANCE_CONFIG_FIELDS+['time']).reverse()
```

Comment:
Just want to make sure, we only care about the recent 7 days' data right?

Reply:
I'm just limiting it that way because I thought it would make the query more efficient. Basically, we want the last occurrence which would almost certainly be in the last 7 days.